### PR TITLE
Stability and color fixes M5Atom client

### PR DESCRIPTION
**Pull request by cyberblaststudios**
- Added reconnect logic that will automatically try to re-connect the M5Atom listener to the server if the socket experiences a disconnect event. This is ideal for people that leave the listeners powered on 24/7, so that they don't have to cycle power on M5Atom devices that may be hard to reach if the server ever goes down.
- Added the ability to use a static IP address, mask, and gateway in the code if the user wishes to do so.
- Fixed color formatting issue for the M5Atom devices. M5Atom deals with color in GRB format, instead of RGB, and the server will send RGB hex values to the clients, so code has to compensate for this on the M5Atom.

### CODE REVIEW

**Comment by abourget** [comment]
Oh well should have checked this first :) Although we're 2 days apart. Just submitted https://github.com/josephdadams/TallyArbiter/pull/338
Will comment inline.

File: listener_clients/M5AtomMatrix-listener/tallyarbiter-m5atom/tallyarbiter-m5atom.ino [hunk 1]
```diff
@@ -320,6 +331,7 @@ void evaluateMode() {
     actualColor.replace("#", "");
     String hexstring = actualColor;
     long colorNumber = (long) strtol( &hexstring[1], NULL, 16);
+ // This order is to compensate for Matrix needing grb.
```
Comment (abourget) [inline listener_clients/M5AtomMatrix-listener/tallyarbiter-m5atom/tallyarbiter-m5atom.ino:334]: oh! I see.. but it didn't seem to change anything here.  Using TallyArbiter 3.0, I needed to change the colors in the Settings for the different Busses, as those are the ones that seem to be used.

**Comment by josephdadams** [comment]
@abourget would you like me to merge this and then you can contribute any remaining necessary changes? Or would you like to ask @cyberblaststudios for write access to his branch to have just one single PR?

File: listener_clients/M5AtomMatrix-listener/tallyarbiter-m5atom/tallyarbiter-m5atom.ino (hunk 1 above)
Comment (wrldtvlr) [reply listener_clients/M5AtomMatrix-listener/tallyarbiter-m5atom/tallyarbiter-m5atom.ino:334]: Hi @abourget ... We've been running multiple Matrices in production for several weeks now and this code seems to work well for us. Let's see if we can figure out what's different with your setup. Starting with the server - are you using the default color assignments for Program and Preview? Second question would be does the web interface on the server reflect the standard color assignments? That is, red for Program and green for Preview?

**Comment by wrldtvlr** [comment]
@josephdadams I'd like to suggest this change be merged into main and be available for the next release as its been working well for us. We would be glad to help troubleshoot any problems people have with the Matrix client going forward.

**Pull request merged**